keps/693-multikueue/README.md

```diff
@@ -223,7 +241,59 @@ If there is enough of global quota, the unknown admitted workloads would be re-a
 the management cluster. If not, some workloads will be preempted to meet the global quota.
 In case of duplicates, all but one of them will be removed.
 
-#### Follow ups ideas
+#### Garbage Collector
+
+Will monitor the Workloads in the remote clusters to detect and remove
+those that ware created by MultiKueue but ware not deleted by the Workload Controller 
+in the normal flow, due to a temporary connectivity outage to the remote cluster.
+
+### Jobs abstraction
+
+In order to work with different types of jobs the MultiKueue Controller will use two abstraction
+interfaces:
+
+```go
+// MultiKueueAdapter interface needed for MultiKueue job delegation.
+type MultiKueueAdapter interface {
+	// Creates the Job object in the worker cluster using remote client, if not already created.
+	// Copy the status from the remote job if already exists.
+	SyncJob(ctx context.Context, localClient client.Client, remoteClient client.Client, key types.NamespacedName, workloadName, origin string) error
+	// Deletes the Job in the worker cluster.
+	DeleteRemoteObject(ctx context.Context, remoteClient client.Client, key types.NamespacedName) error
+	// IsJobManagedByKueue returns:
+	// - a bool indicating if the job object identified by key is managed by kueue and can be delegated.
+	// - a reason indicating why the job is not managed by Kueue
+	// - any API error encountered during the check
+	IsJobManagedByKueue(ctx context.Context, localClient client.Client, key types.NamespacedName, controllerName string) (bool, string, error)
+	// KeepAdmissionCheckPending returns true if the state of the multikueue admission check should be
+	// kept Pending while the job runs in a worker. This might be needed to keep the managers job
+	// suspended and not start the execution locally.
+	KeepAdmissionCheckPending() bool
+	// GVK returns GVK (Group Version Kind) for the job.
+	GVK() schema.GroupVersionKind
+}
+```
+Used by the [Workload Controller](#workload-controller) to interact with the owner of the reconciled workloads.
+
+
+```
+
+// MultiKueueWatcher optional interface that can be implemented by a MultiKueueAdapter
+// to receive job related watch events from the worker cluster.
+// If not implemented, MultiKueue will only receive events related to the job's workload.
+type MultiKueueWatcher interface {
+	// returns an empty list of objects
```

Comment:
what is the purpose of receiving empty list?

Reply:
it's used when starting the watch for the objects, https://github.com/kubernetes-sigs/kueue/blob/e26d2fd418eb8b7bebb3c578d36e5bc18b5440e2/pkg/controller/admissionchecks/multikueue/multikueuecluster.go#L194